Update test base image and move CI to Github workflow

3nids opened the pull request:
Dependency image has been moved to ubuntu focal (this means Qt 5.12)

CI is moved away from Travis to Github workflow
Caching is working (ccache)

former PR #39815

3nids committed:
[quick] fix deprecated use of QApplication::desktopWidget()->screenGeometry()

This takes the first top level window and use it to get the screen.

The "perfect" approach would be to use a widget as argument in the method.

@PeterPetrik thoughts?

3nids committed:
enable test on Ubuntu Focal

3nids committed:
remove defaulting QML

error: QQmlComponent: Component is not ready

3nids committed:
fix docker compose path

3nids committed:
add build ccache

3nids committed:
timeout oracle

3nids committed:
update oracle instant client

3nids committed:
always upload cache

3nids committed:
configure CTEST for GH workflow

3nids committed:
increase cache size

3nids committed:
docker: master=>latest

3nids committed:
install python3-pyproj and protobuf

3nids committed:
increase docker base size

3nids committed:
increase Oracle timeout

3nids committed:
disable other workflows

m-kuhn commented:
Nice!
Was renaming the "stale" workflows done on purpose?

3nids committed:
disable oracle testing

3nids commented:
> Was renaming the "stale" workflows done on purpose?

other tests are disabled for now to avoid wasting resources

3nids commented:
currently failing tests:

ProcessingQgisAlgorithmsTestPt1
ProcessingQgisAlgorithmsTestPt2
ProcessingQgisAlgorithmsTestPt3
ProcessingQgisAlgorithmsTestPt4
ProcessingGdalAlgorithmsRasterTest
ProcessingGdalAlgorithmsVectorTest
ProcessingGrass7AlgorithmsImageryTest
ProcessingGrass7AlgorithmsRasterTest
ProcessingGrass7AlgorithmsVectorTest
ProcessingSagaAlgorithmsTest
qgis_arcgisrestutilstest
qgis_authmanagertest
qgis_coordinatereferencesystemtest
qgis_geometrytest
qgis_geonodeconnectiontest
qgis_maprendererjobtest
qgis_networkaccessmanagertest
qgis_painteffecttest
qgis_pallabelingtest
qgis_ogrproviderguitest
qgis_3drenderingtest
qgis_processingtest
qgis_geometrycheckstest
PyQgsActionManager
PyQgsPythonProvider
PyQgsAnnotation
PyQgsCoordinateOperationWidget
PyQgsGeometryTest
PyQgsMapLayerComboBox
PyQgsMapLayerProxyModel
PyQgsOGRProviderGpkg
PyQgsProcessExecutable
PyQgsQueryResultModel
PyQgsShapefileProvider
PyQgsSvgCache
PyQgsSpatialiteProvider
PyQgsVectorFileWriter
PyQgsVectorLayer
PyQgsVirtualLayerProvider
PyQgsDBManagerGpkg
PyQgsFileDownloader
PyQgsRelationEditWidget
PyQgsAuthManagerPasswordPostgresTest
PyQgsProviderConnectionPostgres
PyQgsOracleProvider
PyQgsServerWMS
PyQgsServerLandingPage
PyQgsServerWMSGetMap
PyQgsServerWMSGetFeatureInfo
PyQgsServerWMSGetPrint
PyQgsServerWMSGetPrintExtra
PyQgsServerWMSDimension
PyQgsAuthManagerPKIOWSTest
PyQgsAuthManagerOAuth2OWSTest

elpaso commented:
@3nids I can take care of the following tests, but it will take some time I guess:


PyQgsFileDownloader
PyQgsAuthManagerPasswordPostgresTest
PyQgsProviderConnectionPostgres
PyQgsServerWMS
PyQgsServerLandingPage
PyQgsServerWMSGetMap
PyQgsServerWMSGetFeatureInfo
PyQgsServerWMSGetPrint
PyQgsServerWMSGetPrintExtra
PyQgsServerWMSDimension
PyQgsAuthManagerPKIOWSTest
PyQgsAuthManagerOAuth2OWSTest

3nids commented:
@elpaso I advice to run this locally with deps Docker image as is done in the workflow https://github.com/qgis/QGIS/blob/test-focal/.github/workflows/run-tests.yml

elpaso commented:
@3nids I'm testing one by one and I've started with `PyQgsPythonProvider` but I cannot make it fail locally, I replicated the docker environment on my machine (I've done it before when a test was failing on Travis so it's not new stuff for me) and I'm testing with  `docker-compose -f .docker/docker-compose-testing.yml run qgis-deps /root/QGIS/.docker/docker-qgis-test.sh` (after adding `-R PyQgsPythonProvider` to ctest command line to execute only this one).

The test passes, and it also passes if I `docker exec -it` and run it manually from inside the docker container.

Full output:

```
$ docker-compose  -p QGIS_workflow_testing  -f .docker/docker-compose-testing.yml run qgis-deps /root/QGIS/.docker/docker-qgis-test.sh
Creating qgis_workflow_testing_qgis-deps_run ... done
Load Postgres database...🐘
Wait a moment while loading PostGreSQL database.
 done 🥩
Restoring postgres test data ...
Restoring tests/testdata/provider/testdata_pg.sql
Restoring tests/testdata/provider/testdata_pg_relations.sql
Restoring tests/testdata/provider/testdata_pg_reltests.sql
Restoring tests/testdata/provider/testdata_pg_role.sql
Restoring tests/testdata/provider/testdata_pg_vectorjoin.sql
Restoring tests/testdata/provider/testdata_pg_hstore.sql
Restoring tests/testdata/provider/testdata_pg_array.sql
Restoring tests/testdata/provider/testdata_pg_raster.sql
Restoring tests/testdata/provider/testdata_pg_topology.sql
Restoring tests/testdata/provider/testdata_pg_domain.sql
Restoring tests/testdata/provider/testdata_pg_json.sql
Restoring tests/testdata/provider/testdata_pg_pointcloud.sql
Restoring tests/testdata/provider/testdata_pg_bigint_pk.sql
Restoring tests/testdata/provider/testdata_pg_hasspatialindex.sql
Database qgis_test populated and ready for use with 'service=qgis_test'
Postgres test data restored ...
Flaky tests are skipped!
List of skipped tests: PyQgsLocalServer|PyQgsPalLabelingServer|qgis_composerutils|PyQgsAppStartup|qgis_spelling|qgis_sipify|qgis_sip_include|qgis_sip_uptodate|PyQgsAuthManagerPKIPostgresTest|PyQgsAuthManagerOgrPostgresTest|PyQgsDbManagerPostgis|qgis_openclutilstest|qgis_layerdefinition|PyQgsProviderConnectionMssql|qgis_filedownloader|qgis_wcsprovidertest|PyQgsWFSProviderGUI|qgis_ziplayertest|PyQgsMssqlProvider|PyQgsPostgresRasterProvider
travis_fold:start:build
Running tests...
Run dashboard with model Experimental
   Source directory: /root/QGIS
   Build directory: /root/QGIS/build
   Reading ctest configuration file: /root/QGIS/CTestConfig.cmake
   Site: github.com
   Build name: (empty)
   Use Experimental tag: 20210130-0848
Build project
   Each symbol represents 1024 bytes of output.
   '!' represents an error and '*' a warning.
    ........................ Size of output: 24K
   0 Compiler errors
   0 Compiler warnings
Test project /root/QGIS/build
    Start 322: PyQgsPythonProvider
1/1 Test #322: PyQgsPythonProvider ..............   Passed    8.69 sec

travis_fold:end:build
100% tests passed, 0 tests failed out of 1

Total Test time (real) =   8.71 sec
```

Am I missing something?

elpaso committed:
Skip tests when TRAVIS=true and add an agnostic QGIS_CONTINUOUS_INTEGRATION_RUN

elpaso commented:
@3nids the tests I've assigned to myself are all passing after https://github.com/qgis/QGIS/pull/41260/commits/3763fc56a9867a59f0fff035492b24bdcb4818bb on my local docker environment.

Let's see how the workflow goes. I just hope we are not switching from bad Travis to worse GitHub... 

If I sum up all the time that I've wasted on Travis nonsense failures during the last years I could have rewritten QGIS in `nodejs` :smile_cat: .

elpaso commented:
@3nids nevermind, I had some deps image leftover, after rebuilding the image I can reproduce most of the issues.

nyalldawson commented:
@3nids where's the appropriate place to set XDG_RUNTIME_DIR ? I think if we correctly set that at /tmp we may resolve a few more issues...

nyalldawson commented:
Looks like some of the failures are coming as a result of using the c locale, which means that QString::localeAwareCompare is case sensitive, not insensitive as the tests expect. Can we hardcode a locale of EN-US on the image?

elpaso commented:
You can try to put locale in the docker env file, it is read by the compose

nyalldawson committed:
Another annotation test mask

nyalldawson committed:
Use unique_ptr

nyalldawson committed:
Use TRANSFER_BACK annotation instead of FACTORY for QgsProviderRegistry::createProvider

Paraphrasing Phil Thomson's advice on https://www.riverbankcomputing.com/pipermail/pyqt/2017-July/039450.html:

"
/Factory/ is used when the instance returned is guaranteed to be new to Python.
In this case it isn't because it has already been seen when being returned by by the python function
creating the provider subclass.

(However for a different sub-class implemented in C++ then it would be the first time it was seen
by Python so the /Factory/ on create() would be correct.)

You might try using /TransferBack/ on createProvider() instead - that might be the best compromise.

nyalldawson committed:
Add a test for python provider using both direct iterator and iterator from layer

nyalldawson committed:
Fix Python based data providers on newer SIP versions

nyalldawson committed:
Rebuild bug_21227-rename-styles.gpkg to fix test on newer GDAL versions

nyalldawson committed:
Move crashing test to its own file

nyalldawson committed:
Fix crash on test exit

nyalldawson committed:
Fix expected results on proj 6+

elpaso committed:
Set US locale for tests

3nids committed:
regroup env vars

3nids committed:
add option to push to cdash

3nids committed:
remove all map layers in PyQgsRelationEditWidget

3nids committed:
set XDG_RUNTIME_DIR

3nids committed:
more tear down

3nids committed:
follow up

elpaso committed:
Don't bark on hana

elpaso committed:
Fix some auth PG tests

elpaso committed:
Fix WMS GFI test references

nyalldawson committed:
When loading GML layers in processing tests, FORCE detection of
SRS. Newer GDAL versions won't do this by default, but we need
to determine the actual/expected CRS for these layers in order
to compare correctly.

nyalldawson commented:
@3nids I think the push to cdash changes you made aren't working properly -- we're no longer getting the results pushed to dash

3nids committed:
fix pushing to CDash

elpaso committed:
Update mask for WMS_GetPrint_StyleDefault

elpaso committed:
Update auth certs readme

3nids committed:
rename method names to be Travis agnostic

elpaso committed:
Re-enable tests and update docs

elpaso committed:
WMS GetMap test images

3nids committed:
fix class methods in TestQgsRelationEditWidget

elpaso committed:
Skip auth password keyring test (no dbus)

elpaso committed:
Fix server WMS control file

3nids committed:
fix dashboard URL

elpaso committed:
Fix server WMS getprint extra control file

3nids committed:
fix tearDown in test QgsActionManager

3nids committed:
try to output CDASH url

elpaso committed:
Fix PG connections API test

elpaso committed:
Update WMS dimensions test control images

3nids committed:
always upload cache for PRs too

elpaso committed:
Fix PKI OWS tests: use TLS 1.2

elpaso committed:
Fix oauh SSL test

3nids committed:
fix cache hit by setting a unique key

elpaso committed:
Re-enable flacky PyQgsPostgresRasterProvider

3nids committed:
add job to show link

3nids committed:
try again to fix relation edit widget test

3nids committed:
fix workflow file

3nids committed:
fix setting output across jobs

elpaso committed:
Fix PyQgsVectorFileWriter test

elpaso committed:
Fix spatialite provider test

elpaso committed:
Fix virtual provider test

elpaso committed:
Fix SHP provider test

elpaso committed:
Fix OGR-GPKG provider test

elpaso committed:
Fix WMS dimensions control file

3nids committed:
set output at proper place

3nids commented:
failing test for 3D rendering shows quite different control images: https://cdash.orfeo-toolbox.org/testDetails.php?test=26523817&build=58951
What shall we do with these?

3nids committed:
post comment with CDASH url when tests are failing

3nids committed:
fix condition

3nids committed:
fix comment action

3nids committed:
set TZ to UTC for Orfeo cdash

3nids committed:
group output for GH workflow (not Travis)

3nids committed:
drop TRAVIS env variable

3nids committed:
more class variable deletion

3nids committed:
write a global comment report

3nids committed:
fix workflow file

3nids committed:
also report when build failed

3nids commented:
updated lists of remaining tests

ProcessingGdalAlgorithmsRasterTest
ProcessingGdalAlgorithmsVectorTest
ProcessingGrass7AlgorithmsImageryTest
ProcessingGrass7AlgorithmsRasterTest
ProcessingGrass7AlgorithmsVectorTest
ProcessingQgisAlgorithmsTestPt1
ProcessingQgisAlgorithmsTestPt2
ProcessingQgisAlgorithmsTestPt3
ProcessingQgisAlgorithmsTestPt4
ProcessingSagaAlgorithmsTest
PyQgsCoordinateOperationWidget
PyQgsGeometryTest
PyQgsGoogleMapsGeocoder
PyQgsOracleProvider
PyQgsRelationEditWidget
PyQgsServerLandingPage
PyQgsServerWMSDimension
PyQgsServerWMSGetMap
qgis_3drenderingtest
qgis_coordinatereferencesystemtest
qgis_geometrycheckstest
qgis_geometrytest
qgis_pallabelingtest

+ Oracle has bee disabled

3nids committed:
fix + more grouping

3nids committed:
fix grouping again

3nids committed:
remove condition

3nids committed:
always run comment job

3nids committed:
correctly report job status

3nids committed:
fix bash conditions

3nids committed:
don't take it too seriously

3nids committed:
bring back conditions

3nids committed:
just a tad more serious

nyalldawson commented:
Rebased to get fix for flaky tile manager test

nyalldawson committed:
Disable geometry z value transform tests on proj 6+, because QGIS
currently doesn't support compound CRS definitions and these are
required for vertical transforms on proj 6+

nyalldawson committed:
Rework failing test for better debug output

nyalldawson committed:
Remove multipolygon is simple test

The behavior in GEOS relating to this was changed in GEOS 3.8
(see https://github.com/libgeos/geos/commit/dddda853c784296ba9f3742e1659e8140c0e8a33)

In any case, we have enough other tests for this method to trust
that we're correctly delegating to GEOS to do the actual test.

nyalldawson committed:
Remove another test for polygon simplicity which depends on GEOS version

nyalldawson committed:
Adapt test results depending on GEOS version

nyalldawson commented:
@3nids the output from test_qgscoordinatereferencesystem.cpp is always truncated (even in the raw github action logs) and I can't see what's actually failing there. Do you know why this is occurring?

nyalldawson committed:
Make TestQgsCoordinateOperationWidget tolerant to presence of grid shift files

nyalldawson committed:
[processing] Add option to ignore part order when comparing geometries to expected results

nyalldawson committed:
More tolerance in Processing QGIS algorithms pt 1 test

nyalldawson committed:
Fix expected test results

nyalldawson committed:
Add another reference hash

nyalldawson committed:
Return invalid variant instead of null variant when feature has no geometry

Otherwise the sip wrapper around the returned variant does NOT
return true to QVariant::isNull() when run from outside QgsApplication.

Anyway, it's more sane and brings this function into line with
the other expression functions which return QVariants() instead
of usertype null QVariants

3nids commented:
> the output from test_qgscoordinatereferencesystem.cpp is always truncated (even in the raw github action logs) and I can't see what's actually failing there. Do you know why this is occurring?

I had a look and I don't see any reason why the log would be truncated. We can try not to use the ctest2.ci python script, to see if this helps?

nyalldawson committed:
Fix running tests where the same layer is used for multiple parameters

nyalldawson committed:
Fix GeometryCollect aggregate detection of geometry result type when running from external python applications

3nids committed:
move all CI logic to ctest2ci

nyalldawson committed:
Allow casting date/datetime values to string to compare to expected results

nyalldawson committed:
Fix some more processing alg tests

nyalldawson committed:
Fix some saga tests

3nids committed:
fix regexes

nyalldawson committed:
Update test

nyalldawson committed:
Update some other tests for qt 5.15

nyalldawson committed:
Bit less noise in Find Projection algorithm

nyalldawson committed:
Temporarily disable the Find Projection test, which is very slow to run on proj 6+ builds

nyalldawson committed:
Fix passing geometry compare options to checkLayersEqual

nyalldawson committed:
Add secondary reference layer for tapered buffer alg

nyalldawson committed:
More tolerance in some tests

nyalldawson committed:
A bit of repairs to vector tiles algorithm test

elpaso committed:
GetMap control imgage (again)

elpaso committed:
Update landing page reference files

3nids committed:
don't run the builds on the branch anymore

elpaso committed:
Update dimensions control file (again)

elpaso committed:
Update dimensions control file (again and again)

elpaso committed:
Fix fromGeos returning 0 for x/y on empty point

nyalldawson committed:
Clear up some clang warnigns in qgsgeos.cpp

nyalldawson committed:
Fix incorrect logic in QgsGeometryDuplicateCheck

nyalldawson committed:
Return nullptr instead of empty point -- better maintains API compatibility with earlier QGIS 3.x releases

Empty points were only introduced recently, and it's possible existing
plugin code may check if the return value is None (not an empty point)

nyalldawson committed:
No need to compare result from GDAL contour alg -- we already have lots of tests covering the creation
of the matching GDAL command, and we can trust that GDAL will generate contours correctly

Avoids fragility in the test when underlying GDAL contour algorithm/logic changes

nyalldawson committed:
Add a test hash

nyalldawson committed:
Quotes also need to be escaped in gdal argument values

nyalldawson committed:
Fix comparison of GDAL crs arguments to expected on proj 6+

nyalldawson committed:
Allow for multiple reference text files

nyalldawson committed:
Add reference prj file for proj 6+

nyalldawson committed:
Fix SHAPE_ENCODING argument should not always get escaped

nyalldawson committed:
Trust GDAL gives correct results given that we are confident we are creating the correct command line string

nyalldawson committed:
Update test

nyalldawson committed:
Add test mask

nyalldawson committed:
Add 3d control image variant

nyalldawson committed:
Fix crs of reference dataset

nyalldawson committed:
Add a reference hash

nyalldawson committed:
v.rast.stats refuses to create shapefiles in newer grass versions, so use a different format

nyalldawson committed:
Add grass raster hashes

nyalldawson committed:
On newer GRASS versions i.eb.hsebal01 requires a raster layer for vapourpressureactual, not a numeric value

nyalldawson committed:
Update grass tests

nyalldawson committed:
Disable some SAGA tests, make others more tolerant

Life is too short to deal with SAGA

nyalldawson committed:
More SAGA test nonsense

nyalldawson committed:
Skip test which triggers a slow iterative grind through all known
CRSes on the CI runs

This is too slow to finish on the CI without timing out (and also
is a very pedantic test -- we probably don't strictly even need
to run it locally!)

nyalldawson committed:
Disable the i.pansharpen test -- GRASS on CI raises a python exception when running this tool (not our fault)

nyalldawson committed:
Disable r.rescale test -- cause intermitten segfaults in GRASS on CI (not our fault)

nyalldawson committed:
More tolerance in SAGA tests

nyalldawson committed:
SAGA silliness

nyalldawson committed:
Use a different technique to compare expected vs result directories in processing tests, which allows us to compare the
contents of files using a more flexible approach

nyalldawson committed:
Use QgsRenderChecker to compare png images when comparing processing result
directories

nyalldawson committed:
More saga test tolerance

elpaso committed:
Control image from local docker test run

nyalldawson committed:
SAGA test tolerance

nyalldawson committed:
Check in mask files

3nids committed:
add option (label) to push docker image with QGIS built

nyalldawson committed:
Use Q_GLOBAL_STATIC for QMap storing QgsCoordinateReference system
so that it's correctly torn down before QgsApplication

Because we can't clean up QgsCoordinateReferenceSystem objects
after QgsApplication tear down

Fixes crash on exit of google maps geocoder test

nyalldawson committed:
Disable mesh simplication 3d render test on CI

It randomly renders a blank image

I suspect this is revealing deeper bugs in the 3d offscreen renderer,
but it fails locally for me too (and always has), so it's not caused
by the CI infrastructure move.

This warrants a deeper dive, but it needs to be done by someone
intimately familiar with Qt3D.

3nids committed:
fix docker repo name

nyalldawson committed:
Saga tolerance

3nids committed:
docker login when pushing image

elpaso commented:
PyQgsServerWMSGetMap is passing in my local docker.

```
1/1 Test #686: PyQgsServerWMSGetMap .............   Passed    9.67 sec

::endgroup::
100% tests passed, 0 tests failed out of 1
```

elpaso committed:
Disable WMS getMap 8bit transparency test

I've tried everything without succes: the test
passes on a local docker env which is supposed
to be exactly the same we run on GH workflow
(reassuring, isn't it?).

nyalldawson committed:
SAGA test hash

nyalldawson committed:
Disable correct test

nyalldawson committed:
Another SAGA hash

nyalldawson committed:
Blocklist Oracle test on Github Actions

The Oracle container is not installed, so we can't run these tests

nyalldawson committed:
Disable PyQgsRelationEditWidget -- this test is crashing for
unknown reasons

nyalldawson commented:
Ok, I'm disabling the last two failing tests:

- PyQgsOracleProvider: as noted above we can't get the Oracle container running under Github Actions. @troopa81  this may be something you are able to fix? (as your team is most knowledgeable about this container setup). For now we'll just proceed without CI testing of Oracle.
- PyQgsRelationEditWidget: causes a crash for unknown reasons. This warrants further investigation, but I don't think we should hold back this work further because of this one test. Basically, we are running MANY more tests on GA vs Travis, so the net win far outweighs one new blocklisted test.

@3nids I think we can merge now... (and then deal with the nightmare of backporting this... :scream: )

nyalldawson commented:
@3nids all tests are passing, but now we've got a new error on the test runners stage. Do you know what this is about?

roya0045 commented:
> @3nids all tests are passing, but now we've got a new error on the test runners stage. Do you know what this is about?

https://stackoverflow.com/questions/43099116/error-the-input-device-is-not-a-tty docker parameter

nyalldawson committed:
Revert "Disable PyQgsRelationEditWidget -- this test is crashing for"

This reverts commit c558e92bc67a919f45512bb9cee419352312212d.

nyalldawson committed:
Fix crash in QgsMapToolCapture destructor if tool is deleted as a result of parent canvas deletion

nyalldawson commented:
PyQgsRelationEditWidget -- actually I've fixed the cause of that crash now, so I've re-enabled the test

nirvn commented on .github/workflows/run-tests.yml:
```diff
@@ -0,0 +1,223 @@
+          docker commit qgis_container qgis/qgis-pr-build:PR${GITHUB_PR_NUMBER}
+          docker push qgis/qgis-pr-build:PR${GITHUB_PR_NUMBER}
+
+      - name: run unit tests
```
I'd capitalize 'run' so its consistent with the other section titles:
![image](https://user-images.githubusercontent.com/1728657/107110668-726e5980-687c-11eb-82ee-84701905abe7.png)

3nids committed:
Fix not tty

3nids committed:
Update run-tests.yml

nyalldawson committed:
Temporarily skip QGIS runners test, just so we can get this work merged and Travis disabled...

nyalldawson committed:
Revert "disable other workflows"

This reverts commit 2b6bd1c7f4028b29f2c68a80f93c58d8918ef226.

nyalldawson committed:
Fix shellcheck SC2035 error

nyalldawson committed:
Fix indentation

nyalldawson committed:
Add header to gen_certs.sh

troopa81 commented:
> PyQgsOracleProvider: as noted above we can't get the Oracle container running under Github Actions. @troopa81 this may be something you are able to fix? (as your team is most knowledgeable about this container setup). For now we'll just proceed without CI testing of Oracle.

I agree. I'll take a look into this and will restore the Oracle tests later. Thank you all for this outstanding work.

nirvn commented:
:tada:

lbartoletti commented:
Thanks!

jef-n commented on CMakeLists.txt:
```diff
@@ -532,18 +532,19 @@ set(QML_IMPORT_PATH "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}" CACHE PATH "QML director
+  # Additional test configuration options e.g. max upload size of test report
+  configure_file(
+    "${CMAKE_SOURCE_DIR}/cmake_templates/CTestCustom.cmake.in"
+    "${CMAKE_BINARY_DIR}/CTestCustom.cmake"
```
@3nids hm, does this break submission of the nightlies?  Probably along with something else. For some reason ``CTestCustom.cmake`` is apparently not picked up anymore and submission is tried to a blank ``https://``.  Copying ``CTestCustom.cmake`` back to ``${CMAKE_SOURCE_DIR}`` helps at some places, but not everywhere (or on all branches).  Any hints?

3nids replied:
Submission to the dashboard? Have you defined CTEST_BUILD_NAME?

jef-n replied:
Yes, to the dashboard.   The nightlies set `BUILDNAME` when configuring cmake.

Windows:

https://github.com/qgis/QGIS/blob/570efee7255d9201c460da095421eee2ea4882aa/ms-windows/osgeo4w/package-nightly.cmd#L120

Debian/Ubuntu:

https://github.com/qgis/QGIS/blob/570efee7255d9201c460da095421eee2ea4882aa/debian/rules#L84

And the osgeo4w testing builds work alike:

https://github.com/jef-n/OSGeo4W/blob/16ae5f88c12fe5678ca0c4c7efd2c352b0e20a91/src/qgis-dev/osgeo4w/package.sh#L131

But the results vary between platforms, branches and repos.

jef-n replied:
It started: https://cdash.orfeo-toolbox.org/index.php?project=QGIS&date=2021-02-20

3nids commented:
Adding `-DPUSH_TO_CDASH=true` will push to CDASH.
For the the build name, it now relies on an env var (`CTEST_BUILD_NAME`). I cannot how it used to work with `BUILDNAME`. We could switch this to a cmake var, that would be cleaner?

3nids commented:
is actually the `BUILDNAME` automatically picked up by CTEST?

jef-n commented:
PUSH_TO_CDASH is set.

Not sure how all this works together anymore.  I think it has worked this way since the nightlies were introduced - to run the tests (HF lisboa I think).

BUILDNAME is only referenced in .ci/config.ctest and qgis-test.ctest.example.

ctest(1) says BUILDNAME is the CTest module variable and CTEST_BUILD_NAME the CTest script variable for the BuildName setting for the "CTest submit step".   But I don't see why the buildname would reset the SubmitURL either - as the CTEST_SUBMIT_URL points at the dashboard in CMakeCache.txt and still it tries to upload to "http://".

And the nightlies run the cmake targets and not ctest directly.

3nids commented:
I changed CTestConfig.cmake to be a template file now (.in), which is configured only when PUSH_TO_CDASH is set.
Do you have CTestConfig.cmake in the build dir?

jef-n commented:
Yes.   ouch, I was talking about CTestConfig.cmake all along - but I linked the wrong line (a few lines above PUSH_TO_CDASH part instead of this within).  6e5f4f36166c5a29759f4f78abeab6e9b579e2aa copies CTestConfig.cmake.

3nids commented:
cool, sorry for the troubles!

jef-n commented:
> cool, sorry for the troubles!

unfortunately not cool - as it doesn't help.  It looked like it did locally, but it doesn't help on build machines.   On all branches the final SubmitURL is http:// now - even on master with 6e5f4f3.   

Except for release-3_16 on windows  64bit - where it fails to build (because somebody apparently reordered the include directories and PROJ_INCLUDE_DIR now comes too late to override proj include of the release with proj nightly's includes; but that's a different story), but it is submitted.  It also fails on 32bit, but that's not submitted.

release-3_18 succeeds to build, but also fails to submit.

but master is not built there anymore (because "current" osgeo4w doesn't fulfil the dependencies since the aggressive dependency lift; another different story).  For the "new" osgeo4w I didn't setup nightlies yet (or better put setup workflows and watched it run out of diskspace…; yet another different story).

The same on the qgis2 with the debian/ubuntu builds - `http://` - but without any exceptions (but the last "nightly" round took 3d10h53m14s).

In the previous round of windows builds it was release-3_18 on 64bit that was submitted - one before I fixed an include that broke the build and one after. Odd.

Maybe somehow related to the cmake build tag?   Seems to switch at different points.   Does any of that ring a bell?

jef-n commented:
confused? -v ;)

roya0045 commented:
> confused? -v ;)

It was the closest to a sadface. Would it even be possible to have  some tests to prevent some compound issues? Though on the other hand I doubt such drastic changes should repeat soon enough..

3nids commented:
cmake_templates/CTestConfig.cmake.in is only configured by the main CMakeLists.txt when ENABLE_TESTS is true.
https://github.com/qgis/QGIS/blob/master/CMakeLists.txt#L536-L551
Actually PUSH_TO_CDASH is only defined when tests are enabled. Is this the issue?

Otherwise, have you set CTEST_BUILD_NAME env var or just BUILDNAME cmake var?
